sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -59,7 +61,9 @@ case class AppendData(
     table: NamedRelation,
     query: LogicalPlan,
     writeOptions: Map[String, String],
-    isByName: Boolean) extends V2WriteCommand
+    isByName: Boolean) extends V2WriteCommand {
+  override def withNewTable(t: NamedRelation): AppendData = copy(table = t)
```

Comment:
Can we just make the `table` be a `child` too and not add a special case for it in the Analyzer?

Reply:
IIRC there was a discussion about it before and we decided to not making `table` as a child for writing commands. The same applies to `InsertIntoStatement`.

I can't clearly recall what the reason was, probably because we need to strip `SubqueryAlias` for `table` in the writing commands.

Reply:
one related discussion: https://github.com/apache/spark/pull/21305/files#r206746383

Reply:
Now I recalled. Scan nodes under writing command are special, as they are not for scan, and shoudn't apply things like filter pushdown.